src/main/java/cws/k8s/scheduler/memory/TaskScaler.java

```diff
@@ -0,0 +1,346 @@
+                maxRequest = availRes.getRam();
+            }
+        }
+        log.info("biggest node has maxRequest = {}", NumberFormat.getNumberInstance(Locale.US).format(maxRequest));
+    }
+
+    /**
+     * After a task was finished, this method shall be called to collect the tasks
+     * resource usage
+     * 
+     * @param task
+     */
+    public void afterTaskFinished(Task task) {
+        if (!active) {
+            return;
+        }
+        BigDecimal peakRss;
+        BigDecimal peakVmem;
+        long realtime;
+        // there is no nextflow trace, when the task failed
+        if (task.wasSuccessfullyExecuted()) {
+            peakRss = NfTrace.getNfPeakRss(task);
+            peakVmem = NfTrace.getNfPeakVmem(task);
+            realtime = NfTrace.getNfRealTime(task);
+        } else {
+            peakRss = BigDecimal.ZERO;
+            peakVmem = BigDecimal.ZERO;
+            realtime = 0;
+            // when a task has failed, we put it on the blacklist, so we will not tamper it again
+            this.blacklist.add(task.getConfig().getName());
+        }
+        // @formatter:off
+        Observation o = Observation.builder()
+                .task( task.getConfig().getTask() )
+                .taskName( task.getConfig().getName() )
+                .success( task.wasSuccessfullyExecuted() )
+                .inputSize( task.getInputSize() )
+                .ramRequest( task.getPod().getRequest().getRam() )
+                .peakVmem(peakVmem)
+                .peakRss(peakRss)
+                .realtime(realtime)
+                .node(task.getNode().getName())
+                .build();
+        // @formatter:on
+        log.info("taskWasFinished, observation={}", o);
+        memoryPredictor.addObservation(o);
+        statistics.addObservation(o);
+
+        // Note: this is a workaround, because the SchedulerConfig does not contain the baseDir
+        if (statistics.baseDir == null) {
+            statistics.baseDir = task.getWorkingDir().substring(0, task.getWorkingDir().lastIndexOf("work"));
+        }
+    }
+
+    public synchronized void beforeTasksScheduled(final List<Task> unscheduledTasks) {
+        if (!active) {
+            return;
+        }
+        log.debug("--- unscheduledTasks BEGIN ---");
+        for (Task t : unscheduledTasks) {
+            log.debug("1 unscheduledTask: {} {} {}", t.getConfig().getTask(), t.getConfig().getName(),
+                    t.getPod().getRequest());
+
+            // if task is already blacklisted, don't touch it again
+            if (this.blacklist.contains(t.getConfig().getName())) {
+                continue;
+            }
+            
+            // if task had no memory request set, it cannot be changed
+            BigDecimal taskRequest = t.getPod().getRequest().getRam();
+            if (taskRequest.compareTo(BigDecimal.ZERO) == 0) {
+                log.info("cannot change task {}, because it had no prior requirements", t.toString());
```

Comment:
This is because the patching would fail? Otherwise it would be better for Kubernetes to set some limits.